cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Entity representation for Metadata
+ */
+public class MetadataEntity {
+
+  public static final String NAMESPACE = "namespace";
+  public static final String APPLICATION = "application";
+  public static final String ARTIFACT = "artifact";
+  public static final String VERSION = "version";
+  public static final String DATASET = "dataset";
+  public static final String STREAM = "stream";
+  public static final String VIEW = "view";
+  public static final String KERBEROS_PRINCIPAL = "namespace";
+  public static final String TYPE = "type";
+  public static final String PROGRAM = "namespace";
+
+  private final List<KeyValue> details;
+
+  private MetadataEntity(List<KeyValue> details) {
+    this.details = details;
+  }
+
+  /**
+   * Creates a {@link MetadataEntity} containing the given dsName. Note: This {@link MetadataEntity} is not
+   * equivalent represent of a DatasetId and does not represent a dataset information completely. The namespace to
+   * which the dataset belongs must be added to the returned {@link MetadataEntity} by the client. Alternatively, you
+   * can also construct a {@link MetadataEntity} which is equivalent to DatasetId by providing the namespace using
+   * {@link MetadataEntity#ofDataset(String, String)}.
+   *
+   * @param dsName the name of the dataset
+   * @return {@link MetadataEntity} containing the dataset name
+   */
+  public static MetadataEntity ofDataset(String dsName) {
+    return new MetadataEntity(Collections.singletonList(new KeyValue("dataset", dsName)));
+  }
+
+  /**
+   * Creates a {@link MetadataEntity} containing the given dsName. Note: This {@link MetadataEntity} is not
```

Comment:
The comment is a little confusing. Since this is in api/, users will think they have to add the namespace. I think it would be better to state that, because this does not include the namespace, it will default to the namespace in which this is used.

Reply:
actually this one has the namespace... the comment applies to the previous method that does not have the namespace.